sdks/java/io/hadoop-file-system/src/main/java/org/apache/beam/sdk/io/hdfs/HadoopFileSystem.java

```diff
@@ -189,7 +189,21 @@ public int read(ByteBuffer dst) throws IOException {
       if (closed) {
         throw new IOException("Channel is closed");
       }
-      return inputStream.read(dst);
+      // O length read must be supported
+      int read = 0;
+      // We avoid using the ByteBuffer based read for Hadoop because some FSDataInputStream
+      // implementations are not ByteBufferReadable,
+      // See https://issues.apache.org/jira/browse/HADOOP-14603
+      if (dst.hasArray()){
```

Comment:
nit: add whitespace `){` -> `) {`

Reply:
+1, strange that this one is not part of checkstyle configuration, I will add it